### PR TITLE
Feature/wcwpp 13 store order information

### PR DESCRIPTION
**Pull request by lucapircher**
## TLDR;

- updates the gebrueder weiss sdk
- fixes breaking changes that came in due to changes to the api schema by gebrueder weiss
- handles the success and fullfilled callbacks
- changes the hardcoded language to en-US
- adds an option page for managing the fields in which the order meta data should be stored
- sets default values for these settings during plugin activation

### CODE REVIEW

File: includes/OrderController.php [hunk 1]
```diff
@@ -47,22 +47,53 @@ public function __construct(SettingsRepository $settings, OrderRepository $order
...
+
+    /**
+     * The callback handler for fulfillment callbacks.
+     *
+     * @param WP_REST_Request $request The post request.
+     * @return WP_REST_Response
+     */
+    public function handleFulfillmentCallback(WP_REST_Request $request): WP_REST_Response
+    {
+        $id = $request->get_params()['id'];
```
Comment (philipp-hemmer) [inline includes/OrderController.php:103]: i assume the id will be definitely in the request params

File: includes/Support/WordPress.php [hunk 2]
```diff
@@ -118,4 +118,27 @@ public static function clearScheduledHook(string $hook): void
...
+    {
+        global $wpdb;
+
+        $query = "
+            SELECT DISTINCT($wpdb->postmeta.meta_key) 
+            FROM $wpdb->posts 
+            LEFT JOIN $wpdb->postmeta 
+            ON $wpdb->posts.ID = $wpdb->postmeta.post_id 
+            WHERE $wpdb->posts.post_type = '%s'
+            # exclude hidden meta keys
```
Comment (philipp-hemmer) [inline includes/Support/WordPress.php:138]: if this is sql the # seems not to fit in there properly

File: tests/Unit/LogisticsOrderFactoryTest.php [hunk 3]
```diff
@@ -82,35 +82,42 @@ public function setUp(): void
...
     {
-        $logisticsOrder = $this->logisticsOrderFactory->buildFromWooCommerceOrder($this->createMockOrder());
+        $logisticsOrder = $this->logisticsOrderFactory->buildFromWooCommerceOrder($this->createMockOrder())->getLogisticsOrder();
 
         $logisticsOrder->getLogisticsAddresses();
         $address = $logisticsOrder->getLogisticsAddresses()[0];
 
-        $this->assertSame("consignee", $address->getAddressType());
+        $this->assertSame("CONSIGNEE", $address->getAddressType());
         $this->assertSame("first-name", $address->getAddress()->getName1());
```
Comment (philipp-hemmer) [inline tests/Unit/LogisticsOrderFactoryTest.php:121]: thats an ugly method name, is it coming from wooCommerce?

**Review by philipp-hemmer — APPROVED** [review]

**Review by mattcrn — APPROVED** [review]

File: includes/OrderController.php (hunk 1 above)
Comment (lucapircher) [reply includes/OrderController.php:103]: The ID is in the URL, hence it has to be there.

File: tests/Unit/LogisticsOrderFactoryTest.php (hunk 3 above)
Comment (lucapircher) [reply tests/Unit/LogisticsOrderFactoryTest.php:121]: That method is a generated one from the api cllient.

File: includes/Support/WordPress.php (hunk 2 above)
Comment (lucapircher) [reply includes/Support/WordPress.php:138]: It is SQL and that line is a comment.
https://dev.mysql.com/doc/refman/8.0/en/comments.html

**Pull request merged**